### PR TITLE
[MRG] Speed up `copy_local_genomes.py` with symbolic links

### LINKED ISSUE #138: investigate hard links or symlinks for private genome collections
@mr-eyes asks over on #130:
> [Does making extra copies of private genomes] cause data redundancy? I'd suggest symlinking them in the new directory with new names.

should revisit this. symlinks might work, but I couldn't get them to do so on a quick test.

### PR DESCRIPTION
**Pull request by mr-eyes**
`genome_grist.copy_local_genomes` takes forever to copy and compress FASTA files. This PR resolves #138 by adding the command line option `--sym` to use symbolic links. `--sym` defaults to False, might be changed later after discussing the PR.

Also upgrading private genomes config file to the latest version in docs (`samples` instead of `sample`).

### CODE REVIEW

**Comment by mr-eyes** [comment]
@ctb symlinks are working. I suspect the issue you observed was symlinking relative paths, not absolute paths, maybe?

The problem now after this PR is that the Snakefile always expects gzipped files to process `{{ident}}_genomic.fna.gz`, which might not exist if symlinking uncompressed Fasta files. Does genome-grist really need the Fasta files to be compressed?

**Comment by ctb** [comment]
cool, thanks!

the sticking point for the compression are the `mpileup_wc` and `build_new_consensus_wc` rules, which need to make the files available to bcftools for alignment/pileup purposes. All of the custom code we've written doesn't care about compression or format (because it uses screed). So that could be fixed, but I couldn't think of a simple solution.

Anyhoo, rather than fix the code to figure out if the file is gzipped and then do the right thing, I just decided that compressing files is good and we should be doing it in the first place :)

**Comment by mr-eyes** [comment]
Perfect! The PR is now ready for the review :) Symlinking will only work now if the Fasta files are already compressed.

**Comment by mr-eyes** [comment]
Observed behavior: Using symlinks will work, but the original filename will show in prefetch results, not the new one in the symlink.

**Comment by ctb** [comment]
if it's just in the prefetch CSV `filename` field, that's probably fine! those paths aren't used for anything.

could you separate out the doc change into its own PR, or do you mind if I do? I think that can/should be merged quickly but I might not get to the symlink PR for a bit.

File: genome_grist/copy_local_genomes.py [hunk 1]
```diff
@@ -50,15 +57,28 @@ def main():
...
             with gzip.open(filename) as fp:
                 fp.read(1)
                 is_gzipped = True
 
-        if is_gzipped:
+        destfile = os.path.join(args.output_directory, f"{ident}_genomic.fna.gz")
+        
+        
+        if args.sym and not is_gzipped:
+            print(f"--sym option requires the Fasta files to be gzipped first.", file= sys.stderr)
```
Comment (ctb) [inline genome_grist/copy_local_genomes.py:70]:
```suggestion
            print("--sym option requires the FASTA files to be already gzipped.", file=sys.stderr)
```

File: genome_grist/copy_local_genomes.py [hunk 2]
```diff
@@ -50,15 +57,28 @@ def main():
...
+            sys.exit(1)
+        
+        if args.sym and is_gzipped:
+            print(f"symbolic linking '{filename}' to '{destfile}'")
+            _src = os.path.abspath(filename)
+            _dst = os.path.abspath(destfile)
+            if os.path.islink(_dst):
+                print(f"symlink {_dst} already exist, consider removing it first.", file= sys.stderr)
```
Comment (ctb) [inline genome_grist/copy_local_genomes.py:78]:
```suggestion
                print(f"symlink {_dst} already exists, consider removing it first.", file=sys.stderr)
```

File: genome_grist/copy_local_genomes.py [hunk 3]
```diff
@@ -50,15 +57,28 @@ def main():
...
                 fp.read(1)
                 is_gzipped = True
 
-        if is_gzipped:
+        destfile = os.path.join(args.output_directory, f"{ident}_genomic.fna.gz")
+        
+        
+        if args.sym and not is_gzipped:
+            print(f"--sym option requires the Fasta files to be gzipped first.", file= sys.stderr)
+            sys.exit(1)
```
Comment (ctb) [inline genome_grist/copy_local_genomes.py:71]:
```suggestion
            sys.exit(-1)
```

I usually use -1 for some reason ;)

File: genome_grist/copy_local_genomes.py [hunk 4]
```diff
@@ -17,7 +17,14 @@ def main():
     p.add_argument('genome_files', nargs='+')
     p.add_argument('-o', '--output-csv', required=True)
     p.add_argument('-d', '--output-directory', required=True)
+    p.add_argument('--sym', required=False, action= 'store_true')
     args = p.parse_args()
+    
+    # Create directories if not exist
```
Comment (ctb) [inline genome_grist/copy_local_genomes.py:23]:
```suggestion
    # Create directory if does not exist
```

**Pull request merged**